include controller_manager/controller_manager.h gives Conflict

When compiling , the include of the header `controller_manager/controller_manager.h` conflicts with `hardware_interface/interface_resources.h`. They have this class and structure called `InterfaceResources`, defined in the same name space:
Does someone know how to fix this, apart from renaming one of them ( which is not a viable option )?
Is its something of the compiler that is too restrictive?

````
>> controller_base.h
namespace hardware_interface
{
  class InterfaceResources;
}

>> interface_resources.h
struct InterfaceResources
{
  InterfaceResources() {}

  InterfaceResources(const std::string& hw_iface, const std::set<std::string>& res)
    : hardware_interface(hw_iface),
      resources(res)
  {}

  /** Hardware interface type. */
  std::string hardware_interface;

  /** Resources belonging to the hardware interface. */
  std::set<std::string> resources;
};
```


```
In file included from /opt/ros/kinetic/include/controller_manager/controller_manager.h:36:
In file included from /opt/ros/kinetic/include/controller_manager/controller_spec.h:41:
/opt/ros/kinetic/include/controller_interface/controller_base.h:41:3: error: class 'InterfaceResources' was previously declared as a struct [-Werror,-Wmismatched-tags]
  class InterfaceResources;
  ^
/opt/ros/kinetic/include/hardware_interface/interface_resources.h:43:8: note: previous use is here
struct InterfaceResources
       ^
/opt/ros/kinetic/include/controller_interface/controller_base.h:41:3: note: did you mean struct here?
  class InterfaceResources;
  ^~~~~
  struct
```

--- a/controller_interface/include/controller_interface/controller_base.h
+++ b/controller_interface/include/controller_interface/controller_base.h
@@ -37,11 +37,6 @@
 #include <hardware_interface/robot_hw.h>
 #include <boost/shared_ptr.hpp>
 
-namespace hardware_interface
-{
-  class InterfaceResources;
-}
-
 namespace controller_interface
 {
 

